File: studies/movies/ggplot2/scatter_ols_separate.R
library(tidyverse)

df <- read_csv("data/movies.csv")
ggplot(df) +
  aes(x = `Production Budget`, y = `Worldwide Gross`, color = `MPAA Rating`) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE)

File: galleries/movies/altair/scatter_ols_separate.py
import pandas as pd
import altair as alt

df = pd.read_csv("data/movies.csv")

chart = (
    alt.Chart(df)
    .mark_point()
    .encode(x="Production Budget", y="Worldwide Gross", color="MPAA Rating")
)
chart += chart.transform_regression(
    "Production Budget", "Worldwide Gross", groupby=["MPAA Rating"]
).mark_line()